storm-core/src/jvm/backtype/storm/scheduler/resource/ResourceAwareScheduler.java

```diff
@@ -28,94 +32,270 @@
+                                            this.conf.get(Config.RESOURCE_AWARE_SCHEDULER_EVICTION_STRATEGY), e.getMessage());
+                                    topologySubmitter.moveTopoFromPendingToAttempted(td);
+                                    break;
+                                }
+                            }
+                            boolean madeSpace = false;
+                            try {
+                                //need to re prepare since scheduling state might have been restored
+                                evictionStrategy.prepare(this.topologies, this.cluster, this.userMap, this.nodes);
+                                madeSpace = evictionStrategy.makeSpaceForTopo(td);
+                            } catch (Exception e) {
+                                LOG.error("Exception thrown when running eviction strategy {} to schedule topology {}. No evictions will be done! Error: {} StackTrace: {}"
+                                        , evictionStrategy.getClass().getName(), td.getName(), e.getClass().getName(), Arrays.toString(e.getStackTrace()));
+                                this.restoreCheckpointSchedulingState(schedulingState);
+                                //since state is restored need the update User topologySubmitter to the new User object in userMap
+                                topologySubmitter = this.userMap.get(td.getTopologySubmitter());
+                                topologySubmitter.moveTopoFromPendingToAttempted(td);
+                                break;
+                            }
+                            if (!madeSpace) {
+                                LOG.debug("Could not make space for topo {} will move to attempted", td);
+                                this.restoreCheckpointSchedulingState(schedulingState);
+                                //since state is restored need the update User topologySubmitter to the new User object in userMap
+                                topologySubmitter = this.userMap.get(td.getTopologySubmitter());
+                                topologySubmitter.moveTopoFromPendingToAttempted(td);
+                                this.cluster.setStatus(td.getId(), "Not enough resources to schedule - " + result.getErrorMessage());
+                                break;
+                            }
+                            continue;
+                        } else if (result.getStatus() == SchedulingStatus.FAIL_INVALID_TOPOLOGY) {
+                            this.restoreCheckpointSchedulingState(schedulingState);
+                            //since state is restored need the update User topologySubmitter to the new User object in userMap
+                            topologySubmitter = this.userMap.get(td.getTopologySubmitter());
+                            topologySubmitter.moveTopoFromPendingToInvalid(td, this.cluster);
+                            break;
```

Comment:
nope.  The logic within the else if moves the topology to the invalid queue and the the logic with else moves the topology in the attempted queue